core/scripts/cli/sponsor/create.js

```diff
@@ -0,0 +1,42 @@
+const inquirer = require("inquirer");
+const getDefaultAccount = require("../wallet/getDefaultAccount");
+
+const create = async (web3, artifacts, emp) => {
+  const ExpandedERC20 = artifacts.require("ExpandedERC20");
+  const { toWei, fromWei, toBN } = web3.utils;
+
+  const sponsorAddress = await getDefaultAccount(web3);
+
+  const scalingFactor = toBN(toWei("1"));
+  // TODO: Understand why we need a .rawValue in one case but not the other.
+  const totalPositionCollateral = toBN((await emp.totalPositionCollateral()).rawValue.toString());
+  const totalTokensOutstanding = toBN((await emp.totalTokensOutstanding()).toString());
+  if (totalTokensOutstanding.isZero()) {
+    console.log("Error: can't create initial position with this tool");
+  }
+  const gcr = totalPositionCollateral.mul(scalingFactor).divRound(totalTokensOutstanding);
+
+  const input = await inquirer.prompt({
+    message: "How many tokens to create?",
+    name: "tokensCreated"
+  });
+  const tokens = toWei(input["tokensCreated"]);
+  const collateralNeeded = toBN(tokens)
+    .mul(gcr)
+    .divRound(scalingFactor);
+  console.log("You'll need", fromWei(collateralNeeded), "ETH to borrow tokens");
```

Comment:
Is it worth checking their `collateralToken.balanceOf()` here and stopping&warning them if insufficient? Not opinionated, either way it will fail.

Reply:
We definitely should, but worth doing only after we have the ETH-WETH logic built. I.e., we should check the user's ETH balance not their collateral token balance.

Reply:
ah yes, I guess the code can be condensed into one line once we don't have to check for WETH. Out of curiosity, why are you using ETH right now instead of some arbitrary ERC20 as collateral? Is it more convenient for testing?

Reply:
I am using an arbitrary ERC20 as collateral.

Reply:
> We definitely should, but worth doing only after we have the ETH-WETH logic built. I.e., we should check the user's ETH balance not their collateral token balance.

Then why does this matter?

Reply:
Because the check for ERC20 balance is useless, throw away code. And checking for ETH balance won't check anything right now.

What change would you like to see here?

Reply:
Nothing to do now, worth remembering for the future. I guess I was missing the point that we are hard-coding WETH as the only collateral token to start? (For now, its just ETH bc we don't have ETH-WETH conversion logic yet, right?)

Reply:
I _think_ we're on the same page regarding what the code is doing here, though maybe we're slightly inaccurate in these comments.

Reply:
I'm gonna try to summarize:
- We are building this CLI tool for creating positions with the hard-coded collateral-token of WETH
- We don't currently have WETH-ETH conversion, so for now we're testing with ETH. Since ETH is not an ERC20, we can't check it with `token.balanceOf()`, therefore any balance checks would be throw away code

Is that understanding correct>

Reply:
Almost.

In the final flow, we:
1. Check the user's balance of ETH
2. Convert ETH to WETH
3. Approve the contract to spend WETH
4. Call create

In the current flow, we use an arbitrary ERC20 token as collateral (the smart contract only works with ERC20). The steps are:
1. Approve the contract to spend ERC20 collateral
2. Call create

Your original suggestion was to check balanceOf() on ERC20 collateral. My counter was that checking the balance of an ERC20 is not part of the final flow, so that code would be thrown away.

Reply:
nicely put, that perfectly captures my confusion haha